subsys/sdfw_services/services/suit_service/suit_ipuc.c

```diff
@@ -0,0 +1,195 @@
+ */
+
+#include <suit_plat_err.h>
+#include <suit_plat_decode_util.h>
+#include <suit_ipuc.h>
+#include <stdint.h>
+#include <sdfw/sdfw_services/suit_service.h>
+#include <sdfw/sdfw_services/ssf_client.h>
+#include "suit_service_decode.h"
+#include "suit_service_encode.h"
+#include "suit_service_types.h"
+#include "suit_service_utils.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(suit_ipuc, CONFIG_SSF_SUIT_SERVICE_LOG_LEVEL);
+
+extern const struct ssf_client_srvc suit_srvc;
+
+suit_plat_err_t suit_ipuc_get_count(size_t *count)
+{
+	int err;
+	struct suit_req req = {0};
+	struct suit_rsp rsp = {0};
+	struct suit_get_ipuc_count_rsp *rsp_data = NULL;
+	const uint8_t *rsp_pkt = NULL;
+
+	if (count == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	req.suit_req_msg_choice = SSF_SUIT_REQ_CHOICE(get_ipuc_count);
+	err = ssf_client_send_request(&suit_srvc, &req, &rsp, &rsp_pkt);
+	if (err != 0) {
+		LOG_ERR("ssf_client_send_request failed");
+		return SUIT_PLAT_ERR_IPC;
+	}
+
+	rsp_data = &rsp.SSF_SUIT_RSP(get_ipuc_count);
+	err = rsp_data->SSF_SUIT_RSP_ARG(get_ipuc_count, ret);
+	if (err != SUIT_PLAT_SUCCESS) {
+		ssf_client_decode_done(rsp_pkt);
+		LOG_ERR("ssf_client response return code: %d", err);
+		return err;
+	}
+
+	*count = rsp_data->SSF_SUIT_RSP_ARG(get_ipuc_count, count);
+
+	ssf_client_decode_done(rsp_pkt);
+	return err;
+}
+
+suit_plat_err_t suit_ipuc_get_info(size_t idx, struct zcbor_string *component_id,
+				   suit_manifest_role_t *role)
+{
+	int err;
+	struct suit_req req = {0};
+	struct suit_rsp rsp = {0};
+	struct suit_get_ipuc_info_req *req_data = NULL;
+	struct suit_get_ipuc_info_rsp *rsp_data = NULL;
+	const uint8_t *rsp_pkt = NULL;
+
+	if (component_id == NULL || role == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	if (component_id->value == NULL) {
+		return SUIT_PLAT_ERR_INVAL;
+	}
+
+	req.suit_req_msg_choice = SSF_SUIT_REQ_CHOICE(get_ipuc_info);
+	req_data = &req.SSF_SUIT_REQ(get_ipuc_info);
+	req_data->SSF_SUIT_REQ_ARG(get_ipuc_info, idx) = idx;
+
+	err = ssf_client_send_request(&suit_srvc, &req, &rsp, &rsp_pkt);
+	if (err != 0) {
+		LOG_ERR("ssf_client_send_request failed");
+		return SUIT_PLAT_ERR_IPC;
+	}
+	rsp_data = &rsp.SSF_SUIT_RSP(get_ipuc_info);
+	err = rsp_data->SSF_SUIT_RSP_ARG(get_ipuc_info, ret);
+	if (err != SUIT_PLAT_SUCCESS) {
+		ssf_client_decode_done(rsp_pkt);
+		LOG_ERR("ssf_client response return code: %d", err);
+		return err;
+	}
+
+	struct zcbor_string *src_component_id =
+		&rsp_data->SSF_SUIT_RSP_ARG(get_ipuc_info, component_id);
+
+	if (component_id->len < src_component_id->len) {
+
+		ssf_client_decode_done(rsp_pkt);
+		return SUIT_PLAT_ERR_SIZE;
+	}
+
+	memcpy((void *)component_id->value, src_component_id->value, src_component_id->len);
+	component_id->len = src_component_id->len;
+	*role = rsp_data->SSF_SUIT_RSP_ARG(get_ipuc_info, role);
+	ssf_client_decode_done(rsp_pkt);
+
+#if (defined CONFIG_SUIT_LOG_LEVEL_INF || defined CONFIG_SUIT_LOG_LEVEL_DBG)
```

Comment:
```suggestion
#if defined(CONFIG_SUIT_LOG_LEVEL_INF) || defined(CONFIG_SUIT_LOG_LEVEL_DBG)
```
Apply bracket changes throughout whole PR

Reply:
done